app/models/discussion.rb

```diff
@@ -8,12 +8,16 @@ class Discussion < ActiveRecord::Base
 
   default_scope -> { order('updated_at DESC') }
 
+  after_create :find_or_create_subscriptions
   after_touch :notify_new_comment
 
   private
 
+  def find_or_create_subscriptions
+    Subscriber.for_discussion(self).subscribe(User.all)
+  end
+
   def notify_new_comment
-    subscriptions = Subscriber.for_discussion(self).subscribe(User.all)
-    subscriptions.map(&:new_comment)
+    find_or_create_subscriptions.map(&:new_comment)
   end
 end
```

Comment:
@zamith pay attention to the changes in this file please. This was patched after getting the updates from your branch, and deals with an issue you had. You were assuming subscriptions were only relevant when commenting, and not when creating a discussion.
This was causing a weird bug, which could be reproduced with the following:

1. create a discussion
2. sign out, and sign in as a different user
3. The serializer would look for the subscription. Since it didn't exist, it would be created on the fly
4. You ended up receiving the email after signing in, instead of right after the discussion was created